lib/parachain/collation_protocol.go

```diff
@@ -96,3 +101,83 @@ type CollationSeconded struct {
 func (CollationSeconded) Index() uint {
 	return 4
 }
+
+const MaxCollationMessageSize uint64 = 100 * 1024
+
+type CollationProtocolV1 struct{}
+
+// Type returns CollationMsgType
+func (*CollationProtocolV1) Type() network.MessageType {
+	return network.CollationMsgType
+}
+
+// Hash returns the hash of the CollationProtocolV1
+func (cp *CollationProtocolV1) Hash() (common.Hash, error) {
+	// scale encode each extrinsic
+	encMsg, err := cp.Encode()
+	if err != nil {
+		return common.Hash{}, fmt.Errorf("cannot encode message: %w", err)
+	}
+
+	return common.Blake2bHash(encMsg)
+}
+
+// Encode a collator protocol message using scale encode
+func (cp *CollationProtocolV1) Encode() ([]byte, error) {
+	enc, err := scale.Marshal(*cp)
+	if err != nil {
+		return nil, err
+	}
+	return enc, nil
+}
+
+func decodeCollationMessage(in []byte) (network.NotificationsMessage, error) {
+	collationMessage := CollationProtocolV1{}
+
+	err := scale.Unmarshal(in, &collationMessage)
+	if err != nil {
+		return nil, fmt.Errorf("cannot decode message: %w", err)
+	}
+
+	return &collationMessage, nil
+}
+
+func handleCollationMessage(peerID peer.ID, msg network.NotificationsMessage) (bool, error) {
+	// TODO: Add things
+	fmt.Println("We got a collation message", msg)
+	return false, nil
+}
+
+func getCollatorHandshake() (network.Handshake, error) {
+	return &collatorHandshake{}, nil
+}
+
+func decodeCollatorHandshake(_ []byte) (network.Handshake, error) {
```

Comment:
Are these functions to be filled in later?

Reply:
no this one is supposed to stay as it is. There are some handshakes that are empty. collator handshake is one of such empty handshake.

Reply:
ah I see, thank you for explaining!